### PR TITLE
fix: prevent panic errors in the validation service

### LINKED ISSUE #611: Validation Service Panic Error.
When provided with this profile, the validation service panics: https://gist.githubusercontent.com/geoffturk/b7ea475aa972ac9ba9dbc90f68f4ab67/raw/041e587a64a533788c93ac37e8e456eaac231829/catalist2.json

### PR DESCRIPTION
**Pull request by maxsarratt**
Resolves #611

### Root Cause:

The application was trying to access index 1 of an empty array. This occurred in the loop where errors are being appended. The issue specifically arises when sources array is empty, yet the code attempts to access sources[i][0] and sources[i][1]

```go
for i := 0; i < len(titles); i++ {
		if len(details) != 0 && len(sources) != 0 {
			errors = append(errors, Error{
				Status: status[i],
				Title:  titles[i],
				Detail: details[i],
				Source: map[string]string{
					sources[i][0]: sources[i][1], <----------------
				},
			})
```

### CODE REVIEW

**Review by TingSyuanWang — COMMENTED** [review]
I have a question regarding lines 51-71 in the `pkg/profile/profilevalidator/validationresult.go` file.

In a typical scenario, the arrays `errorMessages`, `details`, `sources`, and `status` should ideally have the same length. If, for instance, we use a condition like `i < len(details)`, it implies that we would potentially hide additional details if there are fewer error messages. In my view, it would be more beneficial to return the complete error message to the user. Any irregularities can be easily identified during this process.

```
for i, errorMessage := range errorMessages {
	var detail string
	var source []string
	var stat int

	// Use details if available.
	if i < len(details) {
		detail = details[i]
	}

	// Use sources if available.
	if i < len(sources) {
		source = sources[i]
	}

	// Use status if available.
	if i < len(status) {
		stat = status[i]
	}

	vr.AppendError(errorMessage, detail, source, stat)
}
```

File: pkg/profile/profilevalidator/validationresult.go [hunk 1]
```diff
@@ -44,21 +44,46 @@ func (vr *ValidationResult) AppendErrors(
...
+		}
+
+		vr.AppendError(errorMessage, detail, source, stat)
 	}
 }
 
 // Merge combines another ValidationResult into the current one.
 func (vr *ValidationResult) Merge(other *ValidationResult) *ValidationResult {
-	if other == nil || other.Valid {
+	if vr == nil || other == nil || other.Valid {
```
Comment (TingSyuanWang) [inline pkg/profile/profilevalidator/validationresult.go:77]: I'm also curious about the rationale behind checking `if vr == nil` at this point. Is there a specific reason for this condition?

Comment (TingSyuanWang) [inline pkg/profile/profilevalidator/validationresult.go:77]: I'm also curious about the reason behind checking if vr == nil at this point. Is there a specific reason for this condition?

Comment (maxsarratt) [reply pkg/profile/profilevalidator/validationresult.go:77]: I'm afraid we will encounter a nil pointer error again, that's why I added the precaution. Haha.

**Comment by maxsarratt** [comment]
There are some issue swith the helm config. Do now merge this PR.

**Review by geoffturk — APPROVED** [review]
LGTM 🚀

**Pull request merged**